js/components/matrix.js

```diff
@@ -0,0 +1,343 @@
+    },
+
+    /**
+     * Returns unique name to the row with combination of name and rowIndex.
+     * @param {Number} rowIndex - a row index
+     * @returns {String} a name unique to the row
+     */
+    getOptionName: function(rowIndex) {
+      return _.kebabCase(this.name + '-' + rowIndex);
+    },
+
+    /**
+     * Click handler for each radio button in matrix
+     * @param {Object} row - a row object which includes id and label
+     * @param {Object} column - a column object which includes id and label
+     * @param {Number} rowIndex - a row index
+     * @param {Number} colIndex - a column index
+     * @returns {undefined}
+     */
+    clickHandler: function(row, column, rowIndex, colIndex) {
+      if (rowIndex >= 0 && colIndex >= 0) {
+        var el = this.getOptionId(rowIndex, colIndex, 'input');
+
+        $('#' + el).prop('checked', true);
+      }
+
+      if (row.id && this.value[row.id]) {
+        delete this.value[row.id];
+      }
+
+      if (this.value[row.label]) {
+        delete this.value[row.label];
+      }
+
+      this.value[row.id || row.label] = column ? column.id || column.label : undefined;
+
+      this.$emit('_input', this.name, this.value);
+    },
+
+    /**
+     * Focus handler for each radio button in matrix
+     * @param {Number} rowIndex - a row index
+     * @param {Number} colIndex - a column index
+     * @returns {undefined}
+     */
+    focusHandler: function(rowIndex, colIndex) {
+      var $vm = this;
+
+      if (rowIndex >= 0 && colIndex >= 0) {
+        var row = this.rowOptions[rowIndex];
+        var col = this.columnOptions[colIndex];
+
+        $('#' + $vm.getOptionId(rowIndex, colIndex, 'input')).focus();
+
+        $('#' + $vm.getOptionId(rowIndex, colIndex, 'span')).focus();
+
+        this.value[row.id || row.label] = col.id || col.label;
+        this.clickHandler(row, col, rowIndex, colIndex);
+      }
+    },
+
+    /**
+     * Set default undefined value for each row if value is blank
+     * @returns {undefined}
+     */
+    setDefaultValue: function() {
+      var $vm = this;
+
+      $vm.value = {};
+
+      _.forEach(this.rowOptions, function(row) {
+        $vm.$set($vm.value, row.id || row.label, undefined);
+      });
+    },
+
+    /**
+     * Set value for each row as per selection and default value
+     * @returns {undefined}
+     */
+    setValue: function() {
+      var $vm = this;
+
+      if ($vm.value === undefined || _.isEmpty($vm.value)) {
+        this.setDefaultValue();
+      } else {
+        if (typeof $vm.value === 'string') {
+          $vm.value = JSON.parse($vm.value);
+        }
+
+        _.forIn($vm.value, function(key, value) {
+          var row = _.find($vm.rowOptions, function(row) {
+            return (_.has(row, 'label') && _.has(row, 'id')) ? row.id === value : row.label === value;
+          });
+
+          var rowIndex = _.findIndex($vm.rowOptions, function(row) {
```

Comment:
Why not just find the index, then you can directly find `row` using `$vm.rowOptions[rowIndex]`?

Reply:
did the changes